web_src/js/components/DashboardRepoList.vue

```diff
@@ -391,7 +391,7 @@ export default defineComponent({
             </div>
           </div>
         </div>
-        <overflow-menu class="ui secondary pointing tabular borderless menu repos-filter">
+        <overflow-menu class="ui secondary pointing tabular borderless menu repos-filter" :active="reposFilter">
```

Comment:
Why? Doesn't seem right.

Reply:
This dynamically passes the [currently selected menu item] to the overflow-menu component to update the [selected] styling of the right-side tippy button.
https://github.com/go-gitea/gitea/blob/8bab94aacb3a17e938a1e8c2bf5daa6a9bd84e9e/web_src/js/webcomponents/overflow-menu.ts#L7
https://github.com/go-gitea/gitea/blob/8bab94aacb3a17e938a1e8c2bf5daa6a9bd84e9e/web_src/js/webcomponents/overflow-menu.ts#L238-L240
https://github.com/go-gitea/gitea/blob/8bab94aacb3a17e938a1e8c2bf5daa6a9bd84e9e/web_src/js/webcomponents/overflow-menu.ts#L19

Reply:
I do not think it is right. Too complex to explain. In short: you can't make a component depend on its parent's behavior. It can't be designed to only work with Vue.

Reply:
Is it not okay even if it's not a strong dependency? The code should execute if the dependency exists and do nothing if it doesn't.

The original intention was to update the styles inside the overflow-menu whenever there are changes to the selected menu item. So I started thinking about how to receive change notifications within the overflow-menu. Then, when I was looking into `customElements.define`, I came across the following line of code, and I implemented it in the same way. 

https://github.com/go-gitea/gitea/blob/946fb78a950d93ec36a36debc57e061089ebd5af/web_src/js/webcomponents/absolute-date.ts#L13

Reply:
That's different. `'date', 'year'` are really from parent.

But an item is action or not, it is from the overflow-menu itself, not from the parent.

Reply:
Could you please take a look at the latest implementation and see if it's appropriate? Thank you.
https://github.com/go-gitea/gitea/pull/34183/commits/be93e1597a6c768fc6c9e09e7280fd6a4030fdf9